src/lib/allocation_tracker.cc

```diff
@@ -304,14 +314,30 @@ DDRes AllocationTracker::push_lost_sample(MPSCRingBufferWriter &writer,
     return {};
   }
 
-  auto *lost_event = reinterpret_cast<LostEvent *>(buffer.data());
-  lost_event->hdr.size = sizeof(LostEvent);
-  lost_event->hdr.misc = 0;
-  lost_event->hdr.type = PERF_RECORD_LOST;
-  lost_event->id = 0;
-  lost_event->lost = lost_count;
+  auto *event = reinterpret_cast<perf_event_lost *>(buffer.data());
+  event->header.size = sizeof(perf_event_lost);
+  event->header.misc = 0;
+  event->header.type = PERF_RECORD_LOST;
+  auto now = PerfClock::now();
+  event->sample_id.time = now.time_since_epoch().count();
+  if (_state.pid == 0) {
+    _state.pid = getpid();
+  }
+  if (tl_state.tid == 0) {
+    tl_state.tid = ddprof::gettid();
+  }
+  event->sample_id.pid = _state.pid;
```

Comment:
TID and PID info are somewhat questionable considering the lost samples can come from previously lost events ?
Any reason to add these ?

Reply:
Accounting of lost events is done per pid, why not report it ?
tid is indeed more questionable, but since I was adding pid, I also added tid...

Reply:
maybe this can be interesting. We could also in time add a table of lost events per TID.